eng/Signing.props

```diff
@@ -1,140 +1,181 @@
-    </ItemGroup>
-
-    <ItemGroup Condition="'$(SignMsiFiles)' == 'true'">
-      <ItemsToSign Include="$(ArtifactsPackagesDir)**/*.msi" />
-      <ItemsToSign Include="$(ArtifactsPackagesDir)**/*.cab" />
-    </ItemGroup>
-
-    <ItemGroup Condition="'$(SignBurnEngineFiles)' == 'true'">
-      <ItemsToSign Include="@(BundleInstallerEngineArtifact)" />
-    </ItemGroup>
-
-    <ItemGroup Condition="'$(SignBurnBundleFiles)' == 'true'">
-      <!-- Sign the bundles, now that the engine is reattached. Avoid re-signing the engine. -->
-      <ItemsToSign
-        Include="@(BundleInstallerExeArtifact)"
-        Exclude="@(BundleInstallerEngineArtifact)" />
-      <!-- Note: wixstdba is internal to the engine bundle and does not get signed. -->
-    </ItemGroup>
-
-    <ItemGroup Condition="'$(SignFinalPackages)' == 'true'">
-      <DownloadedSymbolPackages Include="$(DownloadDirectory)**\*.symbols.nupkg" />
-      <ItemsToSign Include="$(DownloadDirectory)**\*.nupkg" Exclude="@(DownloadedSymbolPackages)" />
-
-      <!-- The cross OS diagnostics symbol packages need to be signed as they are the only packages
-      that have a specific version of assets that are only meant to be indexed in symbol servers.
-      Since only *symbols.nupkg get indexed, and installer doesn't produce these, we need to glob them for signing. -->
-      <ItemsToSign Include="$(DownloadDirectory)**\*CrossOsDiag*.symbols.nupkg" />
-
-      <ItemsToSign Include="$(DownloadDirectory)**\*.deb" />
-      <ItemsToSign Include="$(DownloadDirectory)**\*.rpm" />
-    </ItemGroup>
-
-    <ItemGroup>
-      <ItemsToSign Update="@(ItemsToSign)" Authenticode="$(CertificateId)" />
-    </ItemGroup>
-
-  </Target>
-
   <ItemGroup>
+    <!--
+      Replace the default items to sign with the specific set we want. This allows the build to call
+      Arcade's Sign.proj multiple times for different sets of files as the build progresses.
+    -->
+    <ItemsToSign Remove="@(ItemsToSign)" />
+
+    <!-- Find bundle artifacts, which need multiple stages to fully sign. -->
+    <BundleInstallerEngineArtifact Include="$(ArtifactsPackagesDir)**/*engine.exe" />
+    <BundleInstallerExeArtifact Include="$(ArtifactsPackagesDir)**/*.exe" />
+
+    <!-- apphost and comhost template files are not signed, by design. -->
+    <FileSignInfo Include="apphost.exe;singlefilehost.exe;comhost.dll" CertificateName="None" />
+
+    <!-- Third-party components which should be signed.  -->
+    <FileSignInfo Include="Newtonsoft.Json.dll" CertificateName="3PartySHA2" />
+    <FileSignInfo Include="Mono.Cecil.dll" CertificateName="3PartySHA2" />
+    <FileSignInfo Include="Mono.Cecil.Mdb.dll" CertificateName="3PartySHA2" />
+    <FileSignInfo Include="Mono.Cecil.Pdb.dll" CertificateName="3PartySHA2" />
+    <FileSignInfo Include="Mono.Cecil.Rocks.dll" CertificateName="3PartySHA2" />
+
+    <FileSignInfo Include="mscordaccore.dll" CertificateName="MicrosoftSHA2" />
```

Comment:
I think we hit interesting behavior of signtool here. This says the short name dac signs with sha2, but the long name is microsoft400.

Reply:
Signtool seems to match on exact name and extension. This code would need to explicitly handle the long name DAC by either calculating the name or globbing it.

Reply:
Yes, the match for `FileSignInfo` is the name without path. I had tried at some point to glob the mscordaccore produced and then take the file name. This won't work when signing in packages which is what we want. For that we'd probably need the versioning target to have run and then add something ugly like:
```suggestion
    <FileSignInfo Include="mscordaccore.dll" CertificateName="MicrosoftSHA2" />
    <FileSignInfo Include="mscordaccore_x86_x86_$(FileVersion).dll CertificateName="MicrosoftSHA2" />
    <FileSignInfo Include="mscordaccore_x86_arm_$(FileVersion).dll CertificateName="MicrosoftSHA2" />
    <FileSignInfo Include="mscordaccore_x64_x64_$(FileVersion).dll CertificateName="MicrosoftSHA2" />
    <FileSignInfo Include="mscordaccore_x64_arm64_$(FileVersion).dll CertificateName="MicrosoftSHA2" />
    <FileSignInfo Include="mscordaccore_arm_arm_$(FileVersion).dll CertificateName="MicrosoftSHA2" />
    <FileSignInfo Include="mscordaccore_arm64_arm64_$(FileVersion).dll CertificateName="MicrosoftSHA2" />
```

The interesting behavior I thought I was going to see was that the file contents are the same, but the names were different. One suggested signing with 400, the other with SHA2. I thought SignTool uses the hash of the content to only sign things once, so I was unsure which one it'd pick. However, but it looks like it uses name and hash so this isn't a concern. It just means the long name has always been signed using 400 since 3.1.

Reply:
Signtool uses hash of content + name. So if you have foo.nupkg and foo.symbols.nupkg and they happen to be identical you can make different signing decisions. This is by design.